File: Leetcode/PY/Leetcode643-子数组的最大平均数.py
from typing import List
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        ans = 0
        sum = 0
        for i in range(k):
            sum += nums[i]
        ans = sum / k
        n = len(nums)
        for i in range(k,n):
            sum += (nums[i] - nums[i - k])
            ans = max(ans,sum / k)
        return ans

File: Leetcode/sbgo/Leetcode643-子数组的最大平均数.go
package sbgo

func findMaxAverage(nums []int, k int) float64 {
	sum := 0
	maxsum := 0
	for i := 0; i < k; i++ {
		sum += nums[i]
	}
	maxsum = sum
	n := len(nums)
	for i := k; i < n; i++ {
		sum += (nums[i] - nums[i-k])
		maxsum = max(sum, maxsum)
	}
	ans := float64(maxsum) / float64(k)
	return ans
}
